### DISCUSSION TITLE
Inofficial OpenDTU App (Beta)

### DISCUSSION THREAD

**Discussion by CommanderRedYT** [tbnobody/OpenDTU#1556] [Show and tell]
# Inofficial OpenDTU App

> [!NOTE]
> This topic is **not** intended for tracking issues. If you need help with the app, please open a issue [here](https://github.com/OpenDTU-App/opendtu-react-native/issues).

Hi,

I just wanted to announce that I am currently working on a open source OpenDTU App! It is not a webview, but a real app build using React Native, meaning it is compatible with android and iOS.

As of now, it is still in development. Things like publishing to stores are not a thing yet. However, apk files are build and released using GitHub Actions. You can find the releases here: https://github.com/OpenDTU-App/opendtu-react-native/releases

Currently the following features are implemented:
- Multi-language support (german and english as of now)
- [mDNS autodiscovery](https://github.com/tbnobody/OpenDTU/pull/1441)
- Multi-OpenDTU support
- Update firmware directly from the app
- View live data
- View some of the stuff from the information tabs
- Database support
  - As of now, only prometheus is supported. You can just set the database url to the instance that is querying the prometheus endpoint of opendtu and it will show simple graphs.
  - It is written in a way where it should easily be possible to add different datasources.

You can find out more and the releases here: https://github.com/OpenDTU-App/opendtu-react-native

I would love to get plenty of feedback, but keep in mind that it is still under heavy development!

## Screenshots
<img src="https://github.com/tbnobody/OpenDTU/assets/43087936/7830a47d-1d54-4f27-b205-093223c10726" alt="Screenshot_1702075762" width=250>

<img src="https://github.com/tbnobody/OpenDTU/assets/43087936/a8c6b365-ddb5-463c-8b04-c81f1ea11d52" alt="Screenshot_1702075874" width=250>

<img src="https://github.com/tbnobody/OpenDTU/assets/43087936/0b2521e9-e857-4679-9801-9eef738dbc87" alt="Screenshot_1702075885" width=250>

<img src="https://github.com/tbnobody/OpenDTU/assets/43087936/5382300b-0f95-4051-89ce-9d3c1af57f78" alt="Screenshot_1702075907" width=250>

<img src="https://github.com/tbnobody/OpenDTU/assets/43087936/1e4a39dc-e6ce-4c70-9541-c192a64c7a52" alt="Screenshot_1702075921" width=250>

**Comment by Marc9485** [comment]
Die Idee find ich gut und ich würde auch helfen aber iOS lässt doch nur Apps aus dem Store zu?!?

**Reply by CommanderRedYT** [reply]
Ja, noch, aber EU macht gerade das Apple Sideloading implementieren muss :)

**Reply by CommanderRedYT** [reply]
Also aktuell sieht das mit den App Stores so aus:
- Google Play Store: 25€ Gebühr fürs Entwickler-Konto erstellen + eine Menge an Daten wie Wohnort etc. zu Google geben
- Apple App Store: Auch wieder teuer, der Entwickler-Account kostet soweit ich weiß 99€/Jahr. Ich hätte Zugang zu einem MacBook, heißt im Emulator und so kann ichs testen, nur veröffentlichen ist halt so ne Sache, da würde ich eher auf das Sideloading warten
- F-Droid: Kenne mich absolut nicht aus mit publishen im F-Droid Store, habs auch schon öfters versucht, imo zu wenig Dokumentation, wenn wer sich besser auskennt bitte helfen :)

**Reply by Tiese** [reply]
@CommanderRedYT: Du hast also eine App entwickelt, die man auf einem iPhone **_nicht_** so einfach installieren kann?!
Ähm, und wie kann ich diese dann benutzen?

**Reply by CommanderRedYT** [reply]
> @CommanderRedYT: Du hast also eine App entwickelt, die man auf einem iPhone **_nicht_** so einfach installieren kann?!
> Ähm, und wie kann ich diese dann benutzen?

Ich habe eine App entwickelt, die in React Native geschrieben ist. Heißt cross-platform. Ist aber wie jede andere App da draußen auch an die Bedingungen der Platform gebunden. Und da Apple leider im Moment 100€/Jahr für ein Entwickler Konto verlangt, ohne das man nichts machen kann, geht's schwer. 

Ich kann mir Mal ein Mac Book ausleihen und die App konfigurieren (wsl noch 2023), und dann geht das so wie hier erklärt: https://reactnative.dev/docs/running-on-device?platform=ios&os=macos

Es gibt aktuell keinen anderen Weg wenn man iOS hat.

Unter Android ist es natürlich easy, da einfach die apk File herunterladen und gut ist. Danke Apple

**Reply by Marc9485** [reply]
Klappt das bei nem Amazon Fire auch? Die laufen ja mein ich auf android-basis nur halt mit eigenem Store

**Reply by ituri** [reply]
Man kann auch ohne das Theater um einen Developer Account die Apps auf seine iOS-Geräte laden, unterliegt halt nur ein paar Restriktionen und ist umständlicher:
https://makaka.org/unity-tutorials/test-ios-app-without-developer-account

**Reply by CommanderRedYT** [reply]
Ich kanns nur im Emulator auf einem MacBook testen, ein iPhone hab ich nicht. Also kann im Moment nur für Android "garantieren" das es geht

**Reply by CommanderRedYT** [reply]
So, also ich hab jetzt alle Vorbereitungen für iOS gemacht. Also jetzt einfach auschecken und mit xcode installieren

**Comment by designermohr** [comment]
Hey. ich bin Entwickler am Mac und auch Besitzer einer OpenDTU. Wenn Du @CommanderRedYT da Hilfe beim Testing etc. benötigst. Melde dich.
Leider ist React nicht mein Fall. Ich kann nur Python/HTML/JavaScript etc.

**Reply by CommanderRedYT** [reply]
Oh, ja das wär ganz cool! Man muss halt bisschen was in XCode aufsetzen. Das Problem ist halt, das im Moment man noch nicht sideloaden kann unter iOS, heißt jeder müsste die App selber compilen.

**Reply by F5BJR** [reply]
How i find prometheus database ?
Thanks

**Reply by CommanderRedYT** [reply]
@F5BJR What exactly do you mean? Prometheus needs to be configured outside of OpenDTU / opendtu-react-native

**Comment by Tobbz2k** [comment]
Cooles Projekt! Könntest du Support für die OpenDTU on Battery hinzufügen? 
https://github.com/helgeerbe/OpenDTU-OnBattery

**Reply by CommanderRedYT** [reply]
Hi, was wäre den dafür notwendig? 

Ich habe sowas selbst nicht, also kann ich das nicht gut testen

**Comment by Tobbz2k** [comment]
Moin, nach welchem Parametern scannst du denn beim Start der App bzw. was
wir geprüft, wenn ich dann eine IP eingebe? Ich schätze mal, da müsste
vielleicht einfach nur der Scan etwas aufgebohrt werden. Ich wollte aber
auch mal prüfen, ob ich nicht einfach mein Gerät mit der normalen openDTU
Version flashen könnte. Bei der on Battery Version gehts darum, dass damit
eine Nulleinspeisung möglich gemacht wurde.

Am Mo., 13. Mai 2024 um 13:25 Uhr schrieb Florian Wetzel <
***@***.***>:

**Reply by CommanderRedYT** [reply]
Ich scanne ganz normal nach mDNS. Die IP wird mithilfe dieser library verifiziert: https://www.npmjs.com/package/ip-regex

Kleine Anmerkung: Ich bin mir nicht sicher ob das Debugging was für diesen Thread ist, bitte eher ein Issue aufmachen bzw. die Antwort-Funktion nutzen.

**Reply by Tobbz2k** [reply]
Thema hat sich erledigt. Gestern hat er die DTU patu nicht gefunden und heute hats über die IP ohne Probleme geklappt. Von daher alles gut und keine Änderungen notwendig ;) Danke für die sehr schnellen Antworten!

**Comment by F5BJR** [comment]
_@F5BJRQue veux-tu dire exactement? Prometheus doit être configuré en dehors d'OpenDTU / opendtu-react-native_

Je suis peut être idiot mais que faut il mettre dans OpenDTU je n' ai pas trouvé d' exemple ?
Qu'est ce que Prométheus? Est ce que cela peut s' installer dans Android ?

**Reply by CommanderRedYT** [reply]
I am not sure if you understood what prometheus is. https://prometheus.io/
There are plenty of tutorials on how to set it up.
Also, please open a issue for problems with the app in the apps repo and not here.

**Comment by CommanderRedYT** [comment]
I am currently enrolling the app into the play store and would need 20 testers as per googles guidelines. If someone wants to participate, please feel free to contact me!

**Comment by Egobo2001** [comment]
Hello.
i just tested your app 0.4.2
When i select ''Inverter'' the app close  itself...

**Comment by Marc9485** [comment]
Is xDTU Connect in AppStore your App too?

**Reply by CommanderRedYT** [reply]
No, I do not know this app.

**Comment by umrath** [comment]
With the new AltStore available, the iOS problem is no longer an issue within the EU.
Any chance to get the app deployed there?

**Reply by CommanderRedYT** [reply]
Will have a look into this later today!

**Reply by heinzgruber** [reply]
> With the new AltStore available, the iOS problem is no longer an issue within the EU. Any chance to get the app deployed there?

100% that would be awesome! Thanks.

**Reply by CommanderRedYT** [reply]
Unfortunately this will not be easy as it still requires the paid apple account...
